### PR TITLE
[Merged by Bors] - fix: setOption linter also recognises a bare `profiler` option

### PR DESCRIPTION
**Pull request by grunweg**
Fix a missing edge case in the linter, when an option name
had only one component (as in the profiler option), and add a test.

At the same time, rewrite the linter to use the `name` API (much nicer!), and perform two small clean-ups to the test file.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Linter/Style.lean [hunk 1]
```diff
@@ -60,9 +60,9 @@ def setOptionLinter : Linter where run := withSetOptionIn fun stx => do
       return
     if let some (head) := stx.find? is_set_option then
       if let some (name) := parse_set_option head then
-        -- Drop a leading backtick.
-        let name := (toString name).drop 1
-        if name.startsWith "pp." || name.startsWith "profiler." || name.startsWith "trace." then
+        -- Drop a leading backtick and determine the first component of the option name.
+        let firstComponent := (((toString name).drop 1).splitOn ".").get! 0
+        if #["pp", "profiler", "trace"].contains firstComponent then
```
Comment (eric-wieser) [inline Mathlib/Tactic/Linter/Style.lean:65]:
```suggestion
        if #[`pp, `profiler, `trace].contains name.getRoot then
```

File: test/LintStyle.lean [hunk 2]
```diff
@@ -10,11 +10,11 @@ section setOption
 -- because the linter is run on `#guard_msgs` itself. This is a known issue, see e.g.
 -- https://leanprover.zulipchat.com/#narrow/stream/348111-batteries/topic/unreachableTactic.20linter.20not.20suppressed.20by.20.60.23guard_msgs.60
 -- We jump through an extra hoop here to silence the warning.
+set_option linter.setOption false
```
Comment (eric-wieser) [inline test/LintStyle.lean:13]: Can you add a comment explaining this line?

Comment (eric-wieser) [reply test/LintStyle.lean:13]: Ah nevermind, the comment was already there!

**Review by eric-wieser — APPROVED** [review]
bors d+

I think the string-processing that caused the bug can be skipped entirely?

File: Mathlib/Tactic/Linter/Style.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Tactic/Linter/Style.lean:65]: Thanks for the pointer! Sadly, this doesn't work, as `parse_set_option` returns an `Ident`: I haven't been able to make it return a `name` instead.

Comment (eric-wieser) [reply Mathlib/Tactic/Linter/Style.lean:65]: Isn't there an `Ident.getName`?

Comment (grunweg) [reply Mathlib/Tactic/Linter/Style.lean]: Thanks for the pointer; I got it to work!

File: Mathlib/Tactic/Linter/Style.lean [hunk 3]
```diff
@@ -31,11 +31,11 @@ namespace Style.SetOption
 
 /-- Whether a syntax element is a `set_option` command, tactic or term:
 Return the name of the option being set, if any. -/
-def parse_set_option : Syntax → Option (Ident)
+def parse_set_option : Syntax → Option (Name)
```
Comment (eric-wieser) [inline Mathlib/Tactic/Linter/Style.lean:34]:
```suggestion
def parse_set_option : Syntax → Option Name
```

**Review by eric-wieser — APPROVED** [review]
bors d+

**Comment by eric-wieser** [comment]
Ah whatever, it's already on the queue, no need to cancel it to remove those parens

File: Mathlib/Tactic/Linter/Style.lean (hunk 3 above)
Comment (grunweg) [reply Mathlib/Tactic/Linter/Style.lean:34]: Right; I've rolled this into #14294.

**Comment by grunweg** [comment]
Thanks for the review!
